tests/unit/rpc/BaseTests.cpp

```diff
@@ -604,3 +604,55 @@ TEST_F(RPCBaseTest, ToLowerModifier)
     ASSERT_TRUE(spec.process(passingInput4));  // empty str no problem
     ASSERT_EQ(passingInput4.at("str").as_string(), "");
 }
+
+TEST_F(RPCBaseTest, ToNumberModifier)
+{
+    auto spec = RpcSpec{
+        {"str", ToNumber{}},
+    };
+
+    auto passingInput = json::parse(R"({ "str": [] })");
+    ASSERT_TRUE(spec.process(passingInput));
```

Comment:
I think, it is better to check that `passingInput` didn't change after this.

Also probably better to split this test in parametrised tests, because currently `passingInput` is shared and multiple tests cases could affect each other.